lib/honeybadger.ex

```diff
@@ -174,13 +174,30 @@ defmodule Honeybadger do
           Honeybadger.notify(exception)
       end
 
+  Send a notification directly from a string, which will be sent as a
+  `RuntimeError`:
+
+      iex> Honeybadger.notify("custom error message")
+      :ok
+
+  Send a notification as a `class` and `message`:
+
+      iex> Honeybadger.notify(%{class: "SpecialError", message: "custom message"})
+      :ok
+
+  Send a notification as a `badarg` atom:
+
+      iex> Honeybadger.notify(:badarg, %{})
+      :ok
+
   If desired additional metadata can be provided as well:
 
-      Honeybadger.notify(%MyException{}, %{culprit_id: 123})
-      #=> :ok
+      iex> Honeybadger.notify(%RuntimeError{}, %{culprit_id: 123})
+      :ok
   """
-  @spec notify(Exception.t, Map.t, list()) :: :ok
-  def notify(exception, metadata \\ %{}, stacktrace \\ nil) do
+  @spec notify(Notice.noticeable, Map.t, list) :: :ok
+  def notify(exception, metadata \\ %{}, stacktrace \\ nil)
+      when is_map(exception) or is_binary(exception) or is_atom(exception) do
```

Comment:
The `notify` function allowed all erlang terms before this change. If we restrict this to just maps, binaries and atoms, we may inadvertently break other's code.

As of now, when you pass a string or iodata or integer or any erlang term, the call gets forwarded to the `Notice.new` function at https://github.com/honeybadger-io/honeybadger-elixir/blob/master/lib/honeybadger/notice.ex#L19 . This further calls the `exception = Exception.normalize(:error, exception)` function which allows the `exception` variable to have any kind of erlang term.

Explicitly documenting the fact that we accept a string is a good thing, we should probably drop the guard clause.

Reply:
Excellent point! The guard clause has been removed.